src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1100,37 +1150,198 @@ private void PopulateParamsFromRestRequest(Dictionary<string, object?> parameter
+        ///                     }])
+        ///                 {
+        ///                     items{
+        ///                         id
+        ///                         title
+        ///                     }
+        ///                 }</example>
+        private void PopulateMutationEntityAndFieldsToAuthorize(
+            Dictionary<string, HashSet<string>> entityToFields,
+            IInputField schema,
+            string entityName,
+            IMiddlewareContext context,
+            object? parameters)
         {
-            string role = string.Empty;
-            if (context.ContextData.TryGetValue(key: AuthorizationResolver.CLIENT_ROLE_HEADER, out object? value) && value is StringValues stringVals)
+            if (parameters is List<ObjectFieldNode> listOfObjectFieldNode)
             {
-                role = stringVals.ToString();
+                // For the example createbook mutation written above, the object value for `item` is interpreted as a List<ObjectFieldNode> i.e.
+                // all the fields present for item namely- title, reviews, publisher, authors are interpreted as ObjectFieldNode.
+                ProcessObjectFieldNodesForAuthZ(
+                    context: context,
+                    entityName: entityName,
+                    schemaObject: ResolverMiddleware.InputObjectTypeFromIInputField(schema),
+                    fieldNodes: listOfObjectFieldNode,
+                    entityTofields: entityToFields);
             }
-
-            if (string.IsNullOrEmpty(role))
+            else if (parameters is List<IValueNode> listOfIValueNode)
+            {
+                // For the example createbooks mutation written above, the list value for `items` is interpreted as a List<IValueNode>
+                // i.e. items is a list of ObjectValueNode(s).
+                listOfIValueNode.ForEach(iValueNode => PopulateMutationEntityAndFieldsToAuthorize(
+                    entityToFields: entityToFields,
+                    schema: schema,
+                    entityName: entityName,
+                    context: context,
+                    parameters: iValueNode));
+            }
+            else if (parameters is ObjectValueNode objectValueNode)
+            {
+                // For the example createbook mutation written above, the node for publisher field is interpreted as an ObjectValueNode.
+                // Similarly the individual node (elements in the list) for the reviews, authors ListValueNode(s) are also interpreted as ObjectValueNode(s).
+                ProcessObjectFieldNodesForAuthZ(
+                    context: context,
+                    entityName: entityName,
+                    schemaObject: ResolverMiddleware.InputObjectTypeFromIInputField(schema),
+                    fieldNodes: objectValueNode.Fields,
+                    entityTofields: entityToFields);
+            }
+            else if (parameters is ListValueNode listValueNode)
```

Comment:
since this last condition is `else if` and not `else` do you know of other cases that would be hit here resulting in `entityToFields` remaining empty?

Reply:
I don't think there are any other cases. I'll change it to else condition.